### PR TITLE
refactor(friends): consolidate mobile and desktop friend list items

### LINKED ISSUE #5058: consolidate best parts of desktop and mobile friends item

### PR DESCRIPTION
**Pull request by josephmcg**
### What this PR does 📖
- title
- simplify the template by using a Map inside of computed to fetch function, tooltip, and icon
- adjust buttons to be less reliant on colors for accessibility and ui reasons. Feel free to debate
- prevent friends and pending request users from showing up in the user search results
- breakout background to `#app` instead of body for better css modal transitions. all layouts use `#app` so its ok
- standardize 12px gap between user picture and name everywhere

### Which issue(s) this PR fixes 🔨
- Resolve #5058 

### Special notes for reviewers 🗒️
- plan to add quickProfile on any user click, but that's still a bit buggy. it still relies on timeouts, so I want to clean that up before adding it everywhere

### CODE REVIEW

File: components/views/friends/search/Search.vue [hunk 1]
```diff
@@ -67,58 +65,49 @@ export default Vue.extend({
...
-      }
       if (matches.length === 0) {
-        this.error = this.$t('friends.not_found') as string
+        this.error = this.$t('friends.not_found')
+      }
+
+      if (matches.length === 1) {
+        const did = matches[0].did
+        if (did === iridium.id) {
+          this.error = this.$t('friends.self_add')
```
Comment (molimauro) [inline components/views/friends/search/Search.vue:95]: this is never reached i think

**Comment by molimauro** [comment]
Dont know if its related to this but if you right click on a friend in the friendsList and then click to the "goto convo" icon, the context menu is still there after the redirect

**Comment by josephmcg** [comment]
Thanks, I was experimenting with the quickprofile stuff and forgot to remove the stopPropagation after we decided to skip for now

File: components/views/friends/search/Search.vue (hunk 1 above)
Comment (josephmcg) [reply components/views/friends/search/Search.vue:95]: true, will revise

**Review by nathan-power — APPROVED** [review]
LGTM

**Pull request merged**